addons/pkg-npm/common/src/main/java/org/commonjava/indy/pkg/npm/content/NPMStoragePathCalculator.java

```diff
@@ -49,19 +51,47 @@
     @Override
     public String calculateStoragePath( final StoreKey key, final String path )
     {
+
         if ( PKG_TYPE_NPM.equals( key.getPackageType() ) )
         {
+            String pkg = path;
+
+            final String extension = httpMetaOrChecksum( path );
+            if ( extension != null )
+            {
+                pkg = path.substring( 0, path.indexOf( extension ) );
+            }
+
             // This is considering the single path for npm standard like "/jquery"
-            final boolean isSinglePath = path.split( "/" ).length < 2;
+            final boolean isSinglePath = pkg.split( "/" ).length < 2;
             // This is considering the scoped path for npm standard like "/@type/jquery"
-            final boolean isScopedPath = path.startsWith( "@" ) && path.split( "/" ).length < 3;
+            final boolean isScopedPath = pkg.startsWith( "@" ) && pkg.split( "/" ).length < 3;
             if ( isSinglePath || isScopedPath )
             {
-                logger.debug( "Modifying target path: {}, appending '{}'", path, METADATA_NAME );
-                return normalize( path, METADATA_NAME );
+                logger.debug( "Modifying target path: {}, appending '{}', store {}", path, METADATA_NAME, key.toString() );
+                return extension != null ?
+                                normalize( pkg, METADATA_NAME + extension ) : normalize( pkg, METADATA_NAME );
             }
         }
 
         return path;
     }
+
+    private String httpMetaOrChecksum( String path )
+    {
+        if ( path.endsWith( HttpExchangeMetadata.FILE_EXTENSION ) )
```

Comment:
I'd use SpecialPathManager and see if it's in there. I think http-metadata.json files should be listable == false, maybe generated (not sure if that's a property of SpecialPathInfo)...and checksums should have some similar flags set.

Reply:
Ah, I will see if that works, not aware of the SpecialPathManager.